03-single-hop-ipv6-icmp/test_spec03.py

```diff
@@ -0,0 +1,182 @@
+
+
+@pytest.mark.skipif(not bridged(["tap0", "tap1"]),
+                    reason="tap0 and tap1 not bridged")
+# nodes passed to riot_ctrl fixture
+@pytest.mark.parametrize('nodes',
+                         [pytest.param(['native', 'native'])],
+                         indirect=['nodes'])
+def test_task02(riot_ctrl):
+    pinger, pinged = (
+        riot_ctrl(0, APP, Shell, port='tap0'),
+        riot_ctrl(1, APP, Shell, port='tap1'),
+    )
+
+    _, pinged_addr = lladdr(pinged.ifconfig_list())
+    assert pinged_addr.startswith("fe80::")
+
+    res = ping6(pinger, pinged_addr,
+                count=1000, interval=100, packet_size=1024)
+    assert res['stats']['packet_loss'] < 1
+
+    assert pktbuf(pinged).is_empty()
+    assert pktbuf(pinger).is_empty()
+
+
+@pytest.mark.skipif(not bridged(["tap0", "tap1"]),
+                    reason="tap0 and tap1 not bridged")
+# nodes passed to riot_ctrl fixture
+@pytest.mark.parametrize('nodes',
+                         [pytest.param(['native', 'native'])],
+                         indirect=['nodes'])
+def test_task03(riot_ctrl):
+    pinger, pinged = (
+        riot_ctrl(0, APP, Shell, port='tap0'),
+        riot_ctrl(1, APP, Shell, port='tap1'),
+    )
+
+    _, pinged_addr = lladdr(pinged.ifconfig_list())
+    assert pinged_addr.startswith("fe80::")
+
+    res = ping6(pinger, pinged_addr,
+                count=3600, interval=1000, packet_size=1024)
+    assert res['stats']['packet_loss'] < 1
+
+    assert pktbuf(pinged).is_empty()
+    assert pktbuf(pinger).is_empty()
+
+
+@pytest.mark.skipif(not interface_exists("tap0"),
+                    reason="tap0 does not exist")
+@pytest.mark.sudo_only      # ping -f requires root
+@pytest.mark.parametrize('nodes',
+                         [pytest.param(['native'] * TASK05_NODES)],
+                         indirect=['nodes'])
+def test_task04(riot_ctrl, log_nodes):
+    node = riot_ctrl(0, APP, Shell, port='tap0')
+    pingers = [pexpect.replwrap.bash() for _ in range(10)]
+
+    _, pinged_addr = lladdr(node.ifconfig_list())
+    assert pinged_addr.startswith("fe80::")
+    iface = bridge('tap0')
+    pinged_addr += "%{}".format(iface)
+    ping_cmd = get_ping_cmd()
+
+    futures = []
+    for pinger in pingers:
+        if log_nodes:
+            pinger.child.logfile = sys.stdout
+        out = pinger.run_command("{ping_cmd} -f -s 1452 {pinged_addr}"
+                                 # pipe to /dev/null because output can go into
+                                 # MiB of data ;-)
+                                 " 2>&1 > /dev/null"
+                                 .format(ping_cmd=ping_cmd,
+                                         pinged_addr=pinged_addr),
+                                 async_=True, timeout=60 * 60)
+        futures.append(out)
+    timeout_futures(futures, 60 * 60)
+
+    time.sleep(60)
+    assert pktbuf(node).is_empty()
+
+
+@pytest.mark.xfail(reason="See https://github.com/RIOT-OS/RIOT/issues/12565")
```

Comment:
Shouldn't this test require `ping -f` since the interval is 0?

```suggestion
@pytest.mark.sudo_only      # ping -f requires root
@pytest.mark.xfail(reason="See https://github.com/RIOT-OS/RIOT/issues/12565")
```

Reply:
No, this task is multiple `native` nodes pinging one `native` node, not the host force pinging one `native` node (that's task 04). Also the reason why this test requires 11 TAP interfaces ;-).